### ISSUE TITLE
bug(mat-calendar): The calendar contains toggle buttons that do not behave like toggle buttons (Accessibility)

### ISSUE THREAD

**Issue by emipre** [angular/components#30190]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

The mat-calendar component contains a button for each selectable date. When navigating the calendar using a screen reader, each button will be read as "toggle button" indicating that the buttons state should change each time the button is pressed. However the button does not behave like a toggle button. Once the button is toggled it cannot be "untoggled", potentially causing confusion for the user. I believe the `aria-pressed` attribute on the button should be removed to correctly reflect the buttons behaviour. 

This was discovered during a recent accessibility evaluation and they provided the following link: https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html

### Reproduction

Example link: https://material.angular.io/components/datepicker/overview#datepicker-inline-calendar

Steps to reproduce:
1. Use the example link in the browser (I used Chrome)
2. Start the screen reader (I used VoiceOver on MacOS)
3. Navigate to the calendar dates an hear them being read as "toggle button"
4. Realize that you cannot change the pressed state of the toggle button by pressing it multiple times

### Expected Behavior

That the button isn't announced as a toggle button when it doesn't behave like a toggle button.

### Actual Behavior

The button is announced as a toggle button even though it doesn't behave like one.

### Environment

- Angular: 19.0.4
- CDK/Material: 19.0.3
- Browser(s): Chrome (similar result expected in other browsers)
- Operating System: macOS

**Comment by Sepandard** [comment]
Just wanted to chime in, I started working on this issue.

**Comment by crisbeto** [comment]
For reference, this was done intentionally in https://github.com/angular/components/pull/24171. We could remove the `aria-pressed` from deselected dates, but the selected one still gets read out as a toggle button.

**Comment by RiinaKuu** [comment]
Bump.

This also violates [WCAG Success Criterion 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html). Closing the calendar pop-up is a **change of context** when it happens after element **defined as a toggle button** is selected.

If `aria-pressed` was completely removed and getting rid of toggle button definitions, couldn't the selected date just use `aria-selected`?

**Comment by MayaMarjut** [comment]
Hey we are building a product for a public market that must comply with European accessibility standards for information and communication technology. This issue is constantly raised in government audits, could you still consider to change the confusing `aria-pressed` attribute to `aria-selected`?
